src/main/java/org/assertj/core/error/ShouldBeEqual.java

```diff
@@ -106,25 +111,28 @@ public static AssertionErrorFactory shouldBeEqual(Object actual, Object expected
    */
   @Override
   public AssertionError newAssertionError(Description description, Representation representation) {
-    if (actualAndExpectedHaveSameStringRepresentation()) {
+    // only use JUnit error message if comparison strategy was standard, otherwise we need to mention it in the
+    // assertion error message to make it clear to the user it was used.
+    if (comparisonStrategy.isStandard() && !actualAndExpectedHaveSameStringRepresentation()) {
       // Example : actual = 42f and expected = 42d gives actual : "42" and expected : "42" and
```

Comment:
this comment only applies when actual and expected have the same representation, it is not relevant here anymore, it should be moved inside `defaultErrorMessage` at line 153.

Reply:
Yes and no. I'll move it down, but in case they are equal we should not try to use the JUnit 4 `ComparisonFailure` exception.